doc/langref.html.in

```diff
@@ -2893,6 +2893,50 @@ test "switch using enum literals" {
 }
       {#code_end#}
       {#header_close#}
+
+      {#header_open|Non-exhaustive enum#}
+      <p>
+      A Non-exhaustive enum can be created by adding a trailing '_' field.
+      It  must specify a tag type and cannot consume every enumeration value.
+      </p>
+      <p>
+      {#link|@intToEnum#} on a non-exhaustive enum cannot fail.
+      </p>
+      <p>
+      A switch on a non-exhaustive enum can include a '_' prong with the following properties:
```

Comment:
I don't love the `_` prong of the switch, `else` should be sufficient.

Reply:
@daurnimator have a look at the rationale in https://github.com/ziglang/zig/issues/2524#issuecomment-570267686. The `_` is admittedly introducing language complication, however, it's required to be part of non-exhaustive enums in order to not give up the ability to have compile errors for missing enum values.

`_` vs `else` is the difference between API compatibility and ABI compatibility. For non-exhaustive enums, both `_` and `else` make the code ABI compatible. However `_` means the code is ABI compatible but not API compatible with new tags. `_` means you get a compile error when a new tag is added, but you will not get *undefined behavior at runtime* when a tag is added.